Mono.Cecil.Cil/Document.cs

```diff
@@ -62,23 +62,43 @@ public sealed class Document : DebugInformation {
 		}
 
 		public DocumentType Type {
-			get { return (DocumentType) type; }
-			set { type = (byte) value; }
+			get { return type.ToType (); }
+			set { type = value.ToGuid (); }
+		}
+
+		public Guid TypeGuid {
+			get { return type; }
+			set { type = value; }
 		}
 
 		public DocumentHashAlgorithm HashAlgorithm {
-			get { return (DocumentHashAlgorithm) hash_algorithm; }
-			set { hash_algorithm = (byte) value; }
+			get { return hash_algorithm.ToHashAlgorithm (); }
+			set { hash_algorithm = value.ToGuid (); }
+		}
+
+		public Guid HashAlgorithmGuid {
+			get { return hash_algorithm; }
+			set { hash_algorithm = value; }
 		}
 
 		public DocumentLanguage Language {
-			get { return (DocumentLanguage) language; }
-			set { language = (byte) value; }
+			get { return language.ToLanguage (); }
+			set { language = value.ToGuid (); }
+		}
+
+		public Guid LanguageGuid {
+			get { return language; }
+			set { language = value; }
 		}
 
 		public DocumentLanguageVendor LanguageVendor {
-			get { return (DocumentLanguageVendor) language_vendor; }
-			set { language_vendor = (byte) value; }
+			get { return language_vendor.ToVendor (); }
+			set { language_vendor = value.ToGuid (); }
+		}
+
+		public Guid LanguageVendorGuid {
+			get { return language_vendor; }
+			set { language_vendor = value; }
 		}
 
 		public byte [] Hash {
```

Comment:
>public byte [] Hash { [](start = 2, length = 21)

Let's also add byte[] EmbeddedSource { get; set; }

Reply:
@tmat yep. What's that about the has with the start and the length?

Reply:
I'm not sure I understand your question.

Reply:
What does this mean?

![tmat](https://user-images.githubusercontent.com/79284/36915055-2880a41a-1e04-11e8-8644-b1a3160d5669.png)

Reply:
Oh, that's an artifact of CodeFlow PR review tool.

Reply:
🤣

Reply:
I can see how it's very confusing in this context.